addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -87,6 +87,7 @@ const GallerySliderWidget = publicWidget.Widget.extend({
         this.$next = this.$indicator.find('li.o_indicators_right').css('visibility', '');
         var $lis = this.$indicator.find('li[data-slide-to]');
         var nbPerPage = Math.floor(this.$indicator.width() / $lis.first().outerWidth(true)) - 3; // - navigator - 1 to leave some space
+        nbPerPage = nbPerPage >= 0 ? nbPerPage : $lis.length
```

Comment:
Thanks for the fix, few changes needed to make this perfect though:
- No ternary structure for this kind of stuff, use a proper if for the case where you actually need to do something
- Needs a comment: typically the kind of code that would go away in a refactoring if no comment
- Install a linter, there is a missing `;` here, your editor should be screaming :wink: 
- Can't we force the snippet to be visible for the duration of the width computation to actually have the right number of items in that case here too? We do that at some other locations, so there must be a way to do that here too.

Reply:
Thank you for the review, I corrected the issues

> Can't we force the snippet to be visible for the duration of the width computation to actually have the right number of items in that case here too? We do that at some other locations, so there must be a way to do that here too.

I didn't find a clear way to do this in our code, so here there is a second attempt to roughly clone and then delete the object which does the trick, perhaps there might be another better way...